sherpa/astro/ui/tests/test_astro_ui_fluxes.py

```diff
@@ -2210,3 +2209,137 @@ def test_sample_flux_pha_component(idval, make_data_path, clean_astro_ui,
+    # covariance), so this fails.
+    #
+    with pytest.raises(IdentifierErr) as exc:
+        ui.sample_flux(id=idval, bkg_id=1,
+                       lo=1, hi=5, num=niter,
+                       correlated=False)
+
+    assert str(exc.value) == 'model stack is empty'
+
+
+@requires_data
+@requires_fits
+@requires_xspec
+@pytest.mark.parametrize("idval", [1, 2])
+def test_sample_flux_pha_bkg(idval, make_data_path, clean_astro_ui,
+                             hide_logging, reset_seed):
+    """Can we get the flux for the background fit.
+
+    In this case we have a source fit that has also been run,
+    so the error analysis should be fine.
+    """
+
+    ui.set_stat('chi2datavar')
+    ui.set_method('levmar')
+
+    niter = 100
+
+    ui.load_pha(idval, make_data_path('3c273.pi'))
+
+    # Need to ungroup so we can use the mask attribute
+    ui.ungroup(idval)
+    ui.ignore(None, 0.8)
+    ui.ignore(7, None)
+
+    d = ui.get_data(idval)
+    b = ui.get_bkg(idval)
+
+    # We need to do it this way (background first)
+    ui.group_counts(num=20, id=idval, bkg_id=1, tabStops=~b.mask)
+    ui.group_counts(num=20, id=idval, tabStops=~d.mask)
+
+    ui.set_bkg_source(idval, ui.powlaw1d.bpl)
+    ui.set_source(idval, ui.xswabs.gabs * ui.powlaw1d.pl)
+
+    bpl = ui.get_model_component('bpl')
+    bpl.gamma = 0.54
+    bpl.ampl = 6.4e-6
+
+    pl = ui.get_model_component('pl')
+    pl.gamma = 2.0
+    pl.ampl = 2e-4
+
+    gabs = ui.get_model_component('gabs')
+    gabs.nh = 0.05
+    gabs.nh.freeze()
+
+    ui.fit_bkg(idval)
+    ui.fit(idval)
+
+    # We don't try to run covar or get the statistic values
+    # (the API doesn't make that easy for background fits).
+    #
+
+    # Use the same seed for both runs.
+    #
+    np.random.seed(287247)
+    bflux1, bflux2, bvals = ui.sample_flux(id=idval, bkg_id=1,
+                                           lo=1, hi=5, num=niter,
+                                           correlated=False)
+
+    np.random.seed(287247)
+    flux1, flux2, vals = ui.sample_flux(id=idval,
+                                        lo=1, hi=5, num=niter,
+                                        correlated=False)
+
+    assert flux2 == pytest.approx(flux1)
+    assert bflux2 == pytest.approx(bflux1)
+
+    assert np.log10(flux1) == pytest.approx([-12.31348652, -12.28200881, -12.34610975])
+    assert np.log10(bflux1) == pytest.approx([-13.15241991, -12.97986006, -13.31382087])
+
+    # Just check that the flux we've got is close to the one calculated by
+    # calc_energy_flux. This is just to test the values we have checked for
+    # the median value of flux1 and bflux1.
+    #
+    # Note that I am slightly concerned that the background flux has been
+    # calculated using the source response, not the background response.
+    # This is too hard to identify with this dataset. We need a test which
+    # has drastically different background response (nad maybe exposure
```

Comment:
nad -> and

Reply:
@wmclaugh I've commited a fix for this. There are fairly-easy-i'm-sure to make a suggestion that @hamogu has been making, which makes it easy for me to add them automatically, but I'm just not sure how.

Reply:
You make a suggestion with the small icon on the top of the edit bad (next to the H B I for "header" "bold" "italics"). If @DougBurke wants to make use o these suggestions to have himself some typing, he needs to go the GH website and click "apply suggestion". That will create a new commit on the branch with the suggestion. (There is also a way to batch them so you get only one commit for all applied suggestions.). This is a true new commit on your fork and GH action will run on it and it can be merged into master etc. If you want to continue to work on that branch locally, then you have to pull form your branch in your fork first.
(Of course, you can also just look and type any changes we suggest as you've done so far.)